src/components/markdown/Spoiler.tsx

```diff
@@ -1,6 +1,47 @@
-import * as React from 'react';
+import React, { Children } from 'react';
+
+function determineIfSingularCodeBlock(
+  firstName: string,
+  numChildren: number
+): boolean {
+  return firstName == 'LanguageSection' && numChildren == 1;
+}
 
 const Spoiler = ({ children, title }) => {
+  let count = 0;
+  let numChildren = 0;
+  let firstName = 'None';
+  let ogProps;
+
+  React.Children.forEach(children, child => {
+    if (count == 0) {
+      numChildren = React.Children.count(children);
+      firstName = child.props.mdxType;
+      ogProps = child.props;
+    }
+    count++;
+  });
+
+  count = 0;
+  const onlyContainsCode: boolean = determineIfSingularCodeBlock(
+    firstName,
+    numChildren
+  );
+
+  const childrenWithProps = React.Children.map(children, child => {
+    if (count == 0 && onlyContainsCode) {
+      count++;
+      return React.cloneElement(child, {
+        children: ogProps.children,
+        mdxType: 'LanguageSection',
+        originalType: ogProps.originalType,
+        expandable: false,
```

Comment:
I feel like replacing `expandable` with `isCodeBlockExpandable` (or even better refactoring into a context) would be better here. @jeffreymeng thoughts?